src/fs/types.ts

```diff
@@ -2,10 +2,61 @@ import type { CID } from "multiformats/cid"
 
 import { Maybe } from "../common/index.js"
 import { PutResult } from "../components/depot/implementation.js"
-import { Path } from "../path/index.js"
+import { DirectoryPath, DistinctivePath, FilePath, Path } from "../path/index.js"
 import { Ucan } from "../ucan/types.js"
 
 
+// 💾 TOP LEVEL
+// ------------
+
+export type API = Exchange & Persistence & Properties & Posix & Sharing
+
+export interface Exchange {
+  addPublicExchangeKey(): Promise<void>
+  hasPublicExchangeKey(): Promise<boolean>
+}
```

Comment:
Can we make these not part of the `FileSystem.API`? I think they're very similar to the `addSampleData` functions, so perhaps they should be somewhere near that function?